core/async/src/test_utils.rs

```diff
@@ -0,0 +1,36 @@
+use crate::futures::DelayedActionRunner;
+
+type FakeDelayedActionTask<T> =
+    Box<dyn FnOnce(&mut T, &mut dyn DelayedActionRunner<T>) + Send + 'static>;
+
+pub struct FakeDelayedActionRunner<T> {
```

Comment:
nit: Can we have a top level comment for this struct that says it queues all actions till `trigger` is called?

And maybe consider renaming `trigger` to `run_all` or `run_queued_actions`